src/azure-cli/azure/cli/command_modules/containerapp/custom.py

```diff
@@ -5022,11 +5022,16 @@ def set_workload_profile(cmd, resource_group_name, env_name, workload_profile_na
 
 
 def add_workload_profile(cmd, resource_group_name, env_name, workload_profile_name, workload_profile_type=None, min_nodes=None, max_nodes=None):
+    r = None
     try:
         r = ManagedEnvironmentClient.show(cmd=cmd, resource_group_name=resource_group_name, name=env_name)
     except CLIError as e:
         handle_raw_exception(e)
 
+    if r and safe_get(r, "properties", "workloadProfiles") is None:
+        raise ValidationError(f"Cannot add workload profile because the environment doesn't enable workload profile.\n"
+                              f"If you want to use Consumption and Dedicated environment, please create a new one.")
```

Comment:
with --enable-workload-profile?

Reply:
Currently the default value of --enable-workload-profile has been set to `True`, no need to specify it when creating a new environment.